Sources/NIOPosix/BaseSocketChannel.swift

```diff
@@ -140,12 +140,18 @@ private struct SocketChannelLifecycleManager {
                 pipeline.syncOperations.fireChannelUnregistered()
             }
 
+        // TODO: This is to support connect after bind for connected-mode datagram sockets. Might need a better solution for this.
+        // origin: .activated
+        case (.activated, .activate):
```

Comment:
Yeah we may want to rethink this a bit. I wonder if we can construct a system whereby only datagram channel allows this.

Reply:
Yes, this was what I was thinking we might be wanting.

Additionally, we should consider whether we want to support calling `connect(2)` after `connect(2)` which seems to be a valid use case to re-associate a connected datagram socket with a new address.

Reply:
Yeah we should probably allow that.

Reply:
This is now handled in ae1b8d409 with some tests.